src/Features/Coupons.php

```diff
@@ -0,0 +1,175 @@
+<?php
+/**
+ * WooCommerce Marketing > Coupons.
+ *
+ * NOTE: DO NOT edit this file in WooCommerce core, this is generated from woocommerce-admin.
+ *
+ * @package Woocommerce Admin
+ */
+
+namespace Automattic\WooCommerce\Admin\Features;
+
+use Automattic\WooCommerce\Admin\Loader;
+use Automattic\WooCommerce\Admin\Notes\WC_Admin_Notes_Coupon_Page_Moved;
+
+/**
+ * Contains backend logic for the Coupons feature.
+ */
+class Coupons {
+
+	use CouponsMovedTrait;
+
+	/**
+	 * Class instance.
+	 *
+	 * @var Coupons instance
+	 */
+	protected static $instance = null;
+
+	/**
+	 * Get class instance.
+	 */
+	public static function get_instance() {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+		return self::$instance;
+	}
+
+	/**
+	 * Hook into WooCommerce.
+	 */
+	public function __construct() {
+		if ( ! is_admin() ) {
+			return;
+		}
+
+		/**
+		 * Filter to override this feature.
+		 *
+		 * @param bool $disabled Whether the normal coupons are overridden. Defaults to true.
+		 */
+		if ( ! apply_filters( 'woocommerce_admin_coupons', true ) ) {
```

Comment:
WC Admin has some built in handling for features.  We might want to use `Loader::is_feature_enabled` here which also allows filtering of features.

Reply:
@joshuatf Ahh interesting - so just to confirm, there is an extisting filter we get free because of feature gating that people could use to disable this? Could you point me to what that might be. I've missed it.

Reply:
Yep, that's correct!  You can use the `woocommerce_admin_features` filter to add/remove features.

Reply:
Updated